src/coreclr/jit/codegenxarch.cpp

```diff
@@ -4397,6 +4398,37 @@ void CodeGen::genCodeForShift(GenTree* tree)
             inst_RV_SH(ins, size, tree->GetRegNum(), shiftByValue);
         }
     }
+#if defined(TARGET_64BIT)
+    else if (compiler->compOpportunisticallyDependsOn(InstructionSet_BMI2) && tree->OperIs(GT_LSH, GT_RSH, GT_RSZ) &&
```

Comment:
nit: Should probably use `tree->OperIsShift()`, should also do the `compiler->compOpportunisticallyDependsOn(InstructionSet_BMI2)` check last if possible.

The opportunistic check sets a flag in the R2R metadata and ideally we only set that flag if we know we are in a condition where we would actually use the instruction

Reply:
Got it. Made the changes.